transport/src/main/java/io/netty/channel/ChannelPromiseNotifier.java

```diff
@@ -15,15 +15,22 @@
  */
 package io.netty.channel;
 
+import io.netty.util.concurrent.Future;
+import io.netty.util.concurrent.FutureListener;
+import io.netty.util.concurrent.GenericFutureListener;
+import io.netty.util.concurrent.Promise;
 import io.netty.util.concurrent.PromiseNotifier;
+import io.netty.util.internal.PromiseNotificationUtil;
+import io.netty.util.internal.logging.InternalLogger;
+import io.netty.util.internal.logging.InternalLoggerFactory;
 
 /**
  * ChannelFutureListener implementation which takes other {@link ChannelPromise}(s) and notifies them on completion.
  *
  * @deprecated use {@link PromiseNotifier}.
  */
 @Deprecated
-public final class ChannelPromiseNotifier
+public class ChannelPromiseNotifier
```

Comment:
why not final anymore ?

Reply:
Leftover from an earlier approach to the `cascadeChannel` methods.